app/views/messages/_message.html.erb

```diff
@@ -32,14 +35,32 @@
       <% end %>
 
       <%# This div needs to be tightly wrapped around the content_text output because the whitespace formatting is important. %>
-      <div class="whitespace-pre-wrap" data-role="content-text"><%= message.content_text %></div>
+      <div class="whitespace-pre-wrap" data-role="content-text" data-clipboard-target="text"><%= message.content_text %></div>
 
       <div class="flex justify-start invisible gap-2 mt-2 text-gray-600 dark:text-gray-300 group-hover:visible">
         <% if message.user? %>
-          <%= icon "pencil", variant: :outline, size: 18, class: 'hover:text-gray-900 dark:hover:text-white cursor-pointer stroke-2', title: "Edit" %>
+          <%= button_tag type: "button",
+                class: "cursor-pointer hover:text-gray-900 dark:hover:text-white" do
+          %>
+            <%= icon "pencil", variant: :outline, size: 18, title: "Edit" %>
+          <% end %>
         <% else %>
-          <%= icon "clipboard", variant: :outline, size: 18, class: 'hover:text-gray-900 dark:hover:text-white cursor-pointer stroke-2', title: "Copy" %>
-          <%= icon "arrow-path", variant: :outline, size: 18, class: 'hover:text-gray-900 dark:hover:text-white cursor-pointer stroke-2', title: "Regenerate" %>
+          <%= button_tag type: "button",
+                class: "cursor-pointer hover:text-gray-900 dark:hover:text-white",
+                data: { action: %|
+                  clipboard#copy
+                  transition#toggleClassOn
+                  mouseleave->transition#toggleClassOff
+                | } do
+          %>
+            <%= icon "clipboard", variant: :outline, size: 18, title: "Copy", data: { transition_target: "transitionable" } %>
+            <%= icon "check", variant: :outline, size: 18, title: "Copied!", data: { transition_target: "transitionable" }, class: "hidden" %>
```

Comment:
When I tried just the tooltip changing from "Copy" to "Copied!" I felt that the experience was too easy to miss. Since we already have a "transitionable" controller that lets you easily toggle elements, I went ahead and added a check icon which is `hidden` and those to flip upon click.

However, this made me realize that if you were to copy a second time it would start as a check and reverse to the clipboard. We do have a "toggleClassOn" which is a one-way transition so I extended the transitions controller slightly to add a "toggleClassOff".

Now the experience is: click to copy and the icon changes from clipboard to check. When your mouse leaves this icon it invisible flips back to a clipboard icon. That way if you move your mouse back over it to copy again it does the flip to check again.